src/Preg.php

```diff
@@ -25,18 +25,18 @@
 final class Preg
 {
     /**
-     * @template TFlags as int-mask<0, 256, 512>
+     * @template TFlags as int-mask<PREG_OFFSET_CAPTURE, PREG_UNMATCHED_AS_NULL>
      *
-     * @param null|mixed[] $matches
-     * @param TFlags       $flags
+     * @param array<array-key, mixed> $matches
+     * @param TFlags                  $flags
      *
-     * @param-out (TFlags is 256
+     * @param-out (TFlags is PREG_OFFSET_CAPTURE
      *     ? array<array-key, array{string, 0|positive-int}|array{'', -1}>
-     *     : (TFlags is 512
+     *     : (TFlags is PREG_UNMATCHED_AS_NULL
      *         ? array<array-key, string|null>
-     *             : (TFlags is 768
-     *                 ? array<array-key, array{string, 0|positive-int}|array{null, -1}>
-     *                 : array<array-key, string>
+     *         : (TFlags is PREG_OFFSET_CAPTURE|PREG_UNMATCHED_AS_NULL
```

Comment:
```suggestion
     *         : (TFlags is int-mask<PREG_OFFSET_CAPTURE, PREG_UNMATCHED_AS_NULL>
```

Reply:
Adding `768` would be the only `int` over constant in the doc.

> or you need to check for `0` at the 1st ternary (which might be easier to write)

do you mean:
```diff
-@param-out (TFlags is PREG_OFFSET_CAPTURE
+@param-out (TFlags is 0|PREG_OFFSET_CAPTURE
```
?

Reply:
I mainly mean https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7873/commits/fda042041fc7f7e244f5621e284013f1f9f1ceaf - `768` is not `PREG_OFFSET_CAPTURE|PREG_UNMATCHED_AS_NULL`, you need to somehow put that number `768` together but `+` is not supported and `|` has union type meaning...

see https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7873#discussion_r1518844626